### PR TITLE
tests: Fix `tests_network_state_nm.yml` CI failure

### PR DESCRIPTION
**Pull request by liangwen12year**
Since the merge of nmstate commit 'nm dns: Support appending static DNS before dynamic DNS' (https://github.com/nmstate/nmstate/pull/2438), nmstate treats interface with `auto-dns: true` as valid to store DNS and tries to use NetworkManager global DNS as much as possible. However, in NM, `dns=systemd-resolved` does not support global DNS configuration, because systemd-resolved only supports per-interface DNS servers (via the D-Bus API). Therefore, write and validate the DNS configuration in /etc/resolv.conf.

Enhancement:

Reason:

Result:

Issue Tracker Tickets (Jira or BZ if any):

### CODE REVIEW

**Comment by liangwen12year** [comment]
Notice that, I omit the validation on Fedora system, because in Fedora, DNS configuration by default will be pushed to systemd-resolved, and `/etc/resolv.conf` is symlinked to the  `/run/systemd/resolve/stub-resolv.conf`, we may need to remove `/etc/resolv.conf` in the first place in order to validate the DNS settings.

**Comment by liangwen12year** [comment]
[citest]

**Comment by liangwen12year** [comment]
[citest]

**Comment by liangwen12year** [comment]
[citest]

**Comment by liangwen12year** [comment]
[citest]

**Comment by liangwen12year** [comment]
[citest]

**Comment by liangwen12year** [comment]
[citest]

**Comment by liangwen12year** [comment]
[citest]

**Comment by liangwen12year** [comment]
[citest]

File: tests/playbooks/tests_network_state.yml [hunk 1]
```diff
@@ -172,15 +149,10 @@
...
-      changed_when: false
-
-    - name: Check if `/etc/resolv.conf` is generated by NM
-      command: grep "Generated by NetworkManager" /etc/resolv.conf
-      register: generate_by_nm
+    - name: Get the DNS configuration from the file
+        `/run/NetworkManager/no-stub-resolv.conf`
+      command: cat /run/NetworkManager/no-stub-resolv.conf
+      register: no_stub_resolvconf
       ignore_errors: true  # noqa ignore-errors
```
[the remark this answers is not in the document]
Comment (liangwen12year) [reply tests/playbooks/tests_network_state.yml]: I removed it, thanks.

**Comment by liangwen12year** [comment]
[citest]

**Review by richm — APPROVED** [review]
lgtm - test passes on all supported el8 and el9 platforms

**Pull request merged**